pkg/resources/streamlit.go

```diff
@@ -0,0 +1,408 @@
+	if v, ok := d.GetOk("directory_location"); ok {
+		rootLocation = fmt.Sprintf("%s/%s", rootLocation, v)
+	}
+	req := sdk.NewCreateStreamlitRequest(id, rootLocation, d.Get("main_file").(string))
+
+	if v, ok := d.GetOk("query_warehouse"); ok {
+		req.WithQueryWarehouse(sdk.NewAccountObjectIdentifier(v.(string)))
+	}
+	if v, ok := d.GetOk("title"); ok {
+		req.WithTitle(v.(string))
+	}
+	if v, ok := d.GetOk("comment"); ok {
+		req.WithComment(v.(string))
+	}
+	if v, ok := d.GetOk("external_access_integrations"); ok {
+		raw := expandStringList(v.(*schema.Set).List())
+		integrations := make([]sdk.AccountObjectIdentifier, len(raw))
+		for i, v := range raw {
+			integrations[i] = sdk.NewAccountObjectIdentifier(v)
+		}
+		req.WithExternalAccessIntegrations(sdk.ExternalAccessIntegrationsRequest{
+			ExternalAccessIntegrations: integrations,
+		})
+	}
+	if err := client.Streamlits.Create(ctx, req); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeSnowflakeID(id))
+
+	return ReadContextStreamlit(false)(ctx, d, meta)
+}
+
+func ReadContextStreamlit(withExternalChangesMarking bool) schema.ReadContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+		client := meta.(*provider.Context).Client
+		id := helpers.DecodeSnowflakeID(d.Id()).(sdk.SchemaObjectIdentifier)
+
+		streamlit, err := client.Streamlits.ShowByID(ctx, id)
+		if err != nil {
+			if errors.Is(err, sdk.ErrObjectNotFound) {
+				d.SetId("")
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity: diag.Warning,
+						Summary:  "Failed to query streamlit. Marking the resource as removed.",
+						Detail:   fmt.Sprintf("Streamlit name: %s, Err: %s", id.FullyQualifiedName(), err),
+					},
+				}
+			}
+			return diag.FromErr(err)
+		}
+
+		streamlitDetails, err := client.Streamlits.Describe(ctx, id)
+		if err != nil {
+			if errors.Is(err, sdk.ErrObjectNotFound) {
+				d.SetId("")
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity: diag.Warning,
+						Summary:  "Failed to query streamlit properties. Marking the resource as removed.",
+						Detail:   fmt.Sprintf("Streamlit name: %s, Err: %s", id.FullyQualifiedName(), err),
+					},
+				}
+			}
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("name", streamlit.Name); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("schema", streamlit.SchemaName); err != nil {
+			return diag.FromErr(err)
+		}
+		stageId, location, err := helpers.ParseRootLocation(streamlitDetails.RootLocation)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("stage", stageId.FullyQualifiedName()); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("directory_location", location); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("main_file", streamlitDetails.MainFile); err != nil {
+			return diag.FromErr(err)
+		}
+		if withExternalChangesMarking {
```

Comment:
I'm not sure, but this resource looks like it could be done without this `if` and additional stuff (just set everything in read). I had similar case in Roles and NetworkPolicies. You can check them and decide (It would simplify the code).

Reply:
Done.